src/safe_reference.h

```diff
@@ -53,6 +53,12 @@ class safe_reference
         std::weak_ptr<T> impl;
 };
 
+template<typename T>
+constexpr bool operator==( safe_reference<T> const &lhs, safe_reference<T> const &rhs )
+{
+    return lhs && rhs && lhs.get() == rhs.get();
```

Comment:
Two safe references pointing to different items can compare equal after both items have been destroyed.